MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -772,75 +776,63 @@ pointer, and which returns an integer, defined thus:
-memory of ``size`` bytes and set:
-
-- ``ydbbuffer->buf_addr`` to the address of the allocated memory,
-- ``ydbbuffer->len_alloc`` to ``size``, and
-- ``ydbbuffer->len_used`` to zero.
-
-``YDB_BUFFER_USED_TO_STRING(ydbstring, ydbbuffer)`` — With
-``ydbstring`` a pointer to a ``ydb_string_t`` structure and
-``ydbbuffer`` a pointer to a ``ydb_buffer_t`` structure, set:
-
-- ``ydbstring->address=ydbbuffer->buf_addr``, and
-- ``ydb_string->length=ydbbuffer->len_used`` (i.e., no changes to
-  ``ydbbuffer``).
-
-``YDB_STRING_FREE(ydbstring)`` — using `ydb_free()`_ free the memory
-at ``ydbstring->address`` and set ``ydbstring->address``,
-``ydbstring->length`` to zero.
-
-``YDB_STRING_NEW(ydbstring,size)`` — using `ydb_malloc()`_ allocate 
-memory of ``size`` bytes and set:
-
-- ``ydbstring->address`` to the address of the allocated memory, and
-- ``ydbstring->length`` to ``size``.
-
-``YDB_STRING_TO_BUFFER(ydbbuffer, ydbstring, used)`` — With ``ydbbuffer``
-a pointer to a ``ydb_buffer_t`` structure, ``ydbstring`` a pointer to
-a ``ydb_string_t`` structure, and ``used`` an unsigned integer, set:
-
-- ``ydbbuffer->buf_addr=ydbstring->address``,
-- ``ydbbuffer->len_alloc=ydbstring->used``, and
-- ``ydbbuffer->len_used=used`` (i.e., no changes to ``ydbstring``).
-
-``YDB_STRLIT_TO_BUFFER(ydbbuffer, strlit)`` — With ``ydbbuffer`` a
-pointer to a ``ydb_buffer_t`` structure, and ``strlit`` a string
-literal, set:
-
-- ``ydbbuffer->buf_addr`` to the address of ``strlit``, and
-- ``ydbbuffer->len_alloc`` and ``ydbbuffer->len_used`` to the length
-  of the string literal excluding its terminating null character.
-
-``YDB_STRLIT_TO_STRING(ydbstring,strlit)`` — With ``ydbstring`` a
-pointer to a ``ydb_string_t`` structure, and ``strlit`` a string
-literal, set
-
-- ``ydbstring->address`` to the address of ``strlit``, and
-- ``ydbstring->length`` to the length of the string literal excluding
-  its terminating null character.
-
-Note that the addresses of the ``strlit`` string literals set in
-``*ydbbuffer`` by invocations of ``YDB_STRLIT_TO_BUFFER()`` and
-``*ydbstring`` by invocations of ``YDB_STRLIT_TO_STRING()`` are almost
-certainly pointers to read-only sections of memory, and any subsequent
-attempt to modify the contents of ``ydbbuffer->buf_addr`` or
-``ydb_string->address`` will thus result in abnormal process
-termination with segmentation violation (SIG-11) that may be hard to
-troubleshoot.
+``YDB_ASSERT(x)`` – Conditionally include this macro in code for
+debugging and testing purposes. If ``x`` is non-zero, it prints an
+error message on ``stderr`` and generates a core file by calling
+`ydb_fork_n_core()`_.
+
+``YDB_BUFFER_IS_SAME(buffer1, buffer2)`` – Use this macro to test
+whether the memory locations (strings) pointed to by two
+``ydb_buffer_t`` structures have the same content, returning ``FALSE``
+(0) if they differ and a non-zero value if the contents are identical.
+
+``YDB_COPY_BUFFER_TO_BUFFER(source, destination, done)`` – Use this
```

Comment:
All macros that have the "done" parameter do no copy if "done" is set to FALSE. That might help clarifying.